treeherder/seta/analyze_failures.py

```diff
@@ -122,6 +132,7 @@ def get_failures_fixed_by_commit():
                                job_note.job.job_type.name, job_note.job.signature.name))
                 continue
 
+            # we now have a legit fixed-by-commit job failure
             failures[revision_id].append(unique_key(
```

Comment:
here if we do:
if revision_id not in failures:
    failures[revision_id] = []

then I think we can remove the clean_failures stuff below in this patch, as long as in the except clause we ensure we del failures[revision_id]